wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/watcher/server/detectors/ProjectTreeTracker.java

```diff
@@ -154,9 +170,41 @@ public void configureHandler(RequestHandlerConfigurator configurator) {
             .paramsAsDto(params)
             .sendAndSkipResult();
       }
+
+      fireCreatedEventIfIsProject(it);
     };
   }
 
+  private void fireCreatedEventIfIsProject(String path) {
+    // Two or more clients can be subscribed to this handler, so need to skip firing the event
```

Comment:
It would be great if we could avoid such comments making the code self explanatory.